Sources/TuistGenerator/Generator/ConfigGenerator.swift

```diff
@@ -284,35 +285,36 @@ final class ConfigGenerator: ConfigGenerating {
     }
 
     private func deploymentTargetDerivedSettings(target: Target) -> SettingsDictionary {
-        guard let deploymentTarget = target.deploymentTarget else {
-            return [:]
-        }
-
         var settings: SettingsDictionary = [:]
-
-        switch deploymentTarget {
-        case let .iOS(version, devices):
-            var deviceFamilyValues: [Int] = []
-            if devices.contains(.iphone) { deviceFamilyValues.append(1) }
-            if devices.contains(.ipad) { deviceFamilyValues.append(2) }
-
-            settings["TARGETED_DEVICE_FAMILY"] = .string(deviceFamilyValues.map { "\($0)" }.joined(separator: ","))
-            settings["IPHONEOS_DEPLOYMENT_TARGET"] = .string(version)
-
-            if devices.contains(.ipad), devices.contains(.mac) {
-                settings["SUPPORTS_MACCATALYST"] = "YES"
-                settings["DERIVE_MACCATALYST_PRODUCT_BUNDLE_IDENTIFIER"] = "YES"
-            } else {
-                // Unless explicitly specified, when the platform the Product is a framework, these default to YES.
-                settings["SUPPORTS_MACCATALYST"] = "NO"
+        var deviceFamilyValues: [Int] = []
+
+        for deploymentTarget in target.deploymentTargets {
+            switch deploymentTarget {
+            case let .iOS(version, devices):
+                if devices.contains(.iphone) { deviceFamilyValues.append(1) }
+                if devices.contains(.ipad) { deviceFamilyValues.append(2) }
+                
+                settings["IPHONEOS_DEPLOYMENT_TARGET"] = .string(version)
+                
+                if devices.contains(.ipad), devices.contains(.mac) {
+                    settings["SUPPORTS_MACCATALYST"] = "YES"
+                    settings["DERIVE_MACCATALYST_PRODUCT_BUNDLE_IDENTIFIER"] = "YES"
+                } else {
+                    // Unless explicitly specified, when the platform the Product is a framework, these default to YES.
+                    settings["SUPPORTS_MACCATALYST"] = "NO"
+                }
+            case let .macOS(version):
+                settings["MACOSX_DEPLOYMENT_TARGET"] = .string(version)
+            case let .watchOS(version):
+                deviceFamilyValues.append(4)
```

Comment:
What are these integers? maybe we should define an enum to better represent this?

Reply:
1/2/3/4 - iphone/ipad/tv/watch

Reply:
This information is already available in https://github.com/tuist/XcodeProj/blob/main/Sources/XcodeProj/Utils/BuildSettingsProvider.swift#L166 which Tuist uses 

I think it's worth considering an alternate approach where for each platform  we merge / append the default settings.

This also ensure we continue to respect the `defaultSettings` enum where users can opt in our out of Tuist adding any of those

Reply:
I Will fix that